src/hdmf/data_utils.py

```diff
@@ -20,7 +20,7 @@ def append_data(data, arg):
         data.append(arg)
         return data
     elif isinstance(data, np.ndarray):
-        return np.append(data,  np.expand_dims(arg, axis=0), axis=0)
```

Comment:
Why make this change?

`np.append` will flatten multidimensional arrays. The result will always be one-dimensional. `VectorData` columns can have 2-D (or more) data, and with this change, calling `add_row` will flatten those 2-D data arrays into 1-D arrays. I guess since tests pass, we are probably missing a test for this.

```python
>>> a = [[1, 2, 3], [4, 5, 6]]
>>> b = [7, 8, 9]
>>> np.append(a, np.expand_dims(b, axis=0), axis=0)
array([[1, 2, 3],
       [4, 5, 6],
       [7, 8, 9]])
>>> np.append(a, b)
array([1, 2, 3, 4, 5, 6, 7, 8, 9])
```